src/styles/text/text_labels.js

```diff
@@ -319,8 +325,8 @@ export const TextLabels = {
 
         // repeat rules include the text
         if (layout.repeat_distance) {
-            if (text instanceof Array){
```

Comment:
@dmvaldman this was not getting triggered because the individual strings are passed into this function in separate calls, not as an array in a single call (that might have been true in an earlier version).

The change here is a bit messy but improves the repeat logic in some areas.